src/components/google-maps/container.ts

```diff
@@ -0,0 +1,115 @@
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import GoogleMapsApi from 'load-google-maps-api';
+import { getCurrentGeoLocation } from './helpers';
+
+// Require the marker image to show the current position
+const currentPosMarkerImg = require('./current-position-marker.png');
+
+@Component
+export default class GoogleMapContainer extends Vue {
+  // The Google Maps API Key
+  @Prop({
+    type: String,
+    required: false
+  })
+  readonly apiKey!: string;
+
+  // The Google Maps Configuration
+  @Prop({
+    type: Object,
+    required: false,
+    default: () => ({
+      center: { lat: 0, lng: 0 },
+      zoom: 18
+    })
+  })
+  readonly config!: google.maps.MapOptions;
+
+  // The google map object
+  // google: google.maps | null = null;
+  // TODO: fix TS error that doesn't allow google.maps as a type
+  google!: any;
```

Comment:
Not sure, but try `google!: typeof GoogleMapsApi` if its the whole api you are trying to type.

Reply:
or `typeof googlemaps`?

Reply:
Ah no so the issue is that I want to use `google.maps` as the type.

So:
```
google: google.maps;
```

But `google.maps` is just a namespace in the typings not a class or an interface at all. So TS complains about it. I might need to maybe make a PR to change it in the typings or maybe make a wrapper around it? Not sure. Just doesn't seem too beneficial atm. So just added a TODO instead.

Reply:
Actually, just thought of a way to do it. Will update in a sec.

Reply:
` typeof google.maps`?

Reply:
I had to do it recently with typing a namespace...

Reply:
I wasted a whole day trying to solve this damn problem!

Reply:
Yep! idk why i forgot about it until I typed my response to you haha!